crates/swc/tests/fixture/issue-3235/7/output/index.js

```diff
@@ -0,0 +1,18 @@
+import * as swcHelpers from "@swc/helpers";
+import regeneratorRuntime from "regenerator-runtime";
+function foo() {
+    return _foo.apply(this, arguments);
+}
+function _foo() {
+    _foo = swcHelpers.asyncToGenerator(regeneratorRuntime.mark(function _callee() {
+        return regeneratorRuntime.wrap(function _callee$(_ctx) {
+            while(1)switch(_ctx.prev = _ctx.next){
+                case 0:
+                case "end":
+                    return _ctx.stop();
+            }
+        }, _callee);
+    }));
+    return _foo.apply(this, arguments);
+}
+var swcHelpers = 1;
```

Comment:
Hi, @kdy1 ,
I add a test case.
It seems to be a regression in my changes.

Reply:
If two identifier has same symbol and same syntax context, those are equal to each other.
If different variable get a same syntax context, then it's bug of `resolver` pass.

Reply:
in this [REPL](https://play.swc.rs/?version=1.2.124&code=H4sIAAAAAAAAA6vmUgCCnNQShUQFWwVDa65aLq6yxCIoDwAly%2FyBHgAAAA%3D%3D&config=H4sIAAAAAAAAA0WOSw6DMAwFr1J5zaKtShecoJsewkoNTZWfbFMRodydBCGxs57njb3CTwwMKyRkIW6T5KC4wABkPIphmxS6itVoRCdUOlDkibQhcr%2Fe%2Brp2MQodQAfeBjvmJjPRJyaRc4VhcnSqaFHigO5FLhFXTnluuY%2BfuXEraE60n3pCOdWHzsr7AFuv6sJ%2F7%2BwPVhuYL0dPl0cPpWyRCl%2B%2F7QAAAA%3D%3D), we get the wrong result because we can't distinguish which one needs to be renamed.

Reply:
Hmm... Then we may remove the logic for retaining identifiers of top-level items?

Reply:
it is not related to top level. We should not touch the `var` Ident.
See this [swc playground](https://play.swc.rs/?version=1.2.133&code=H4sIAAAAAAAAA6vmUgCCajAJAjmpJQqJCrYKhtYK%2BvoKxRn5pTkpCkWpeYm5qWA1tVxgqiyxCFNZdmpqgQJYJVBVWmlecklmfp5CWn6%2BhibUBqrbAwACzhNBvwAAAA%3D%3D&config=H4sIAAAAAAAAA0WOSw6DMAwFr1J5zaKtShecoJsewkoNTZWfbFMRodydBCGxs57njb3CTwwMKyRkIW6T5KC4wABkPIphmxS6itVoRCdUOlDkibQhcr%2Fe%2Brp2MQodQAfeBjvmJjPRJyaRc4VhcnSqaFHigO5FLhFXTnluuY%2BfuXEraE60n3pCOdWHzsr7AFuv6sJ%2F7%2BwPVhuYL0dPl0cPpWyRCl%2B%2F7QAAAA%3D%3D) and this [Babel REPL](https://babeljs.io/repl#?browsers=chrome%2045&build=&builtIns=false&corejs=false&spec=false&loose=false&code_lz=N4KABBapNgNgUwC5gIZgLxgIwG4wD0BYAzgBYD2ArnACZgBOCAdqgLYLiQC-IXEAN1QM0mHPiKlKNegGsECAA5hWHELxAAzKswDGSAJYVmYTRQoAKAJRR-t2JEQp0WPIWLlqdRi3acYGjBCIi7i7lJecgrKqpzcQA&debug=false&forceAllTransforms=false&shippedProposals=false&circleciRepo=&evaluate=false&fileSize=false&timeTravel=true&sourceType=module&lineWrap=true&presets=env&prettier=true&targets=&version=7.16.9&externalPlugins=&assumptions=%7B%7D).

Reply:
Oh... You are right, although I'm not sure how can we fix this.

Scoping in rust is super-hard because of `no two &mut to same data`.

Reply:
You could check the first commit in this pull request that I added some test cases about it.